core/src/main/scala/org/apache/spark/SparkContext.scala

```diff
@@ -671,7 +674,11 @@ class SparkContext(config: SparkConf) extends Logging with ExecutorAllocationCli
       classOf[LongWritable],
       classOf[BytesWritable],
       conf=conf)
-    val data = br.map{ case (k, v) => v.getBytes}
+    val data = br.map { case (k, v) =>
+      val bytes = v.getBytes
+      assert(bytes.length == recordLength, "Byte array does not have correct length")
+      bytes
```

Comment:
Do you mean something more than these notes we're adding? I just clarified the notes a bit to make it obvious the check is on the byte array.